lib/exercises.rb

```diff
@@ -2,19 +2,51 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  hash = {}
+  acc = 0
+  return_array = []
+  strings.each do |str|
+    if !(hash[str.split(//).sort!.join().hash])
+      hash[str.split(//).sort!.join().hash] = acc
```

Comment:
Since you're reusing `str.split(//).sort!.join().hash`, I suggest storing that in a variable